apps/api/v2/src/ee/bookings/2024-08-13/controllers/bookings.controller.ts

```diff
@@ -0,0 +1,159 @@
+import { Permissions } from "@/modules/auth/decorators/permissions/permissions.decorator";
+import { ApiAuthGuard } from "@/modules/auth/guards/api-auth/api-auth.guard";
+import { PermissionsGuard } from "@/modules/auth/guards/permissions/permissions.guard";
+import {
+  Controller,
+  Post,
+  Logger,
+  Body,
+  UseGuards,
+  Req,
+  Get,
+  Param,
+  Query,
+  BadRequestException,
+  HttpCode,
+  HttpStatus,
+} from "@nestjs/common";
+import { ApiTags as DocsTags } from "@nestjs/swagger";
+import { User } from "@prisma/client";
+import { Request } from "express";
+
+import { BOOKING_READ, BOOKING_WRITE, SUCCESS_STATUS } from "@calcom/platform-constants";
+import {
+  CreateBookingInputPipe,
+  CreateBookingInput,
+  GetBookingsInput_2024_08_13,
+  RescheduleBookingInput_2024_08_13,
+  CancelBookingInput_2024_08_13,
+  MarkAbsentBookingInput_2024_08_13,
+} from "@calcom/platform-types";
+
+@Controller({
+  path: "/v2/bookings",
+  version: VERSION_2024_08_13_VALUE,
+})
+@UseGuards(PermissionsGuard)
+@DocsTags("Bookings")
+export class BookingsController_2024_08_13 {
+  private readonly logger = new Logger("BookingsController");
+
+  constructor(private readonly bookingsService: BookingsService_2024_08_13) {}
+
+  @Post("/")
+  async createBooking(
+    @Body(new CreateBookingInputPipe())
+    body: CreateBookingInput,
+    @Req() request: Request
+  ): Promise<CreateBookingOutput_2024_08_13> {
+    const booking = await this.bookingsService.createBooking(request, body);
+
+    if (Array.isArray(booking)) {
+      await this.bookingsService.billBookings(booking);
+    } else {
+      await this.bookingsService.billBooking(booking);
+    }
+
+    return {
+      status: SUCCESS_STATUS,
+      data: booking,
+    };
+  }
+
+  @Get("/:bookingUid")
+  async getBooking(@Param("bookingUid") bookingUid: string): Promise<GetBookingOutput_2024_08_13> {
+    if (!bookingUid) {
+      throw new BadRequestException("Booking UID is required in request path /:bookingUid");
+    }
+
+    const booking = await this.bookingsService.getBooking(bookingUid);
+
+    return {
+      status: SUCCESS_STATUS,
+      data: booking,
+    };
+  }
+
+  @Get("/")
+  @UseGuards(ApiAuthGuard)
+  @Permissions([BOOKING_READ])
+  async getBookings(
+    @Query() queryParams: GetBookingsInput_2024_08_13,
+    @GetUser() user: User
+  ): Promise<GetBookingsOutput_2024_08_13> {
+    const bookings = await this.bookingsService.getBookings(queryParams, user);
+
+    return {
+      status: SUCCESS_STATUS,
+      data: bookings,
+    };
+  }
+
+  @Post("/:bookingUid/reschedule")
+  async rescheduleBooking(
+    @Param("bookingUid") bookingUid: string,
+    @Body() body: RescheduleBookingInput_2024_08_13,
+    @Req() request: Request
+  ): Promise<RescheduleBookingOutput_2024_08_13> {
+    if (!bookingUid) {
+      throw new BadRequestException("Booking UID is required in request path /:bookingUid/reschedule");
+    }
```

Comment:
shouldn't we create a guard for this ? and use it in all the endpoints which have bookingUid

Reply:
Agree Here

Reply:
Good idea - implemented in [feat: BookingUidGuard](https://github.com/calcom/cal.com/pull/16200/commits/45b593bb27210e5f32367d3077112498c9340690)